Add support for metric filters

sjauld opened the pull request:
# Description

Adds support for inclusive and exclusive metric filters

Fixes #2765

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] Documentation update required

## Checklist:

Please delete options that are not relevant.

- [x] My commit message follows [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] I have performed a self-review of my own code
- [x] I have updated the documentation

## How to test this change?

Add the module using my fork and see what happens

```terraform
module "newrelic" {
  source = "github.com/sjauld/terraform-provider-newrelic//examples/modules/cloud-integrations/aws"

  newrelic_account_id     = "<your account ID>"
  newrelic_account_region = "<your region>"
  name                    = "a prefix for your resources"

  include_metric_filters = {
    "AWS/EC2"    = []
    "AWS/ECS"    = []
    "AWS/Lambda" = []
    "AWS/RDS"    = []
  }
}
```

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/newrelic/terraform-provider-newrelic?pullRequest=2766) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/newrelic/terraform-provider-newrelic?pullRequest=2766) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/newrelic/terraform-provider-newrelic?pullRequest=2766) it.</sub>

pranav-new-relic commented:
@sjauld hi 👋 one tiny suggestion to start with - would you mind updating the documentation here
https://github.com/newrelic/terraform-provider-newrelic/blob/33aa8650ee9bac915494de144be05859ddb3ba83/website/docs/guides/cloud_integrations_guide.html.markdown?plain=1#L50-L73
with a piece of information on the new variables you've added (just a mention of the two, in the list of variables that already exists), followed by an example to explain how one should use these? Thanks!

sjauld commented:
Thanks for the quick feedback @pranav-new-relic. I have updated the docs as suggested.

pranav-new-relic commented on website/docs/guides/cloud_integrations_guide.html.markdown:
```diff
@@ -71,7 +76,8 @@ Variables:
 * newrelic_account_region: The region of your New Relic account, this can be `US` for United States or `EU` for Europe. (Default `US`)
 * name: A unique name used throughout the module to name the resources.
-
+* exclude_metric_filters (optional): a map of namespaces and metric names to exclude from the Cloudwatch metric stream. `Conflicts with include_metric_filters`.
+* include_metric_filters (optional): a map of namespaces and metric names to include in the Cloudwatch metric stream. `Conflicts with exclude_metric_filters`.
```
```suggestion
* `exclude_metric_filters` (Optional): A map of namespaces and metric names to exclude from the Cloudwatch Metric Stream.
* `include_metric_filters` (Optional): A map of namespaces and metric names to include in the Cloudwatch Metric Stream.
  * _Please note that `exclude_metric_filters` and `include_metric_filters` conflict with each other._
```

a tiny cosmetic correction :)

pranav-new-relic commented:
Hey @sjauld, a few asks from us - 

- we're currently unable to test this change fully, owing to unforeseen/intermittent AWS access glitches. While it is evident from the implementation of your change that the _current behaviour of the module_ (as it was, prior to this change) would remain unaffected (i.e. adding these new parameters would not affect current behaviour), can you please test the following scenarios by means of a `terraform apply` just to make sure we're 100% sure this doesn't affect current behaviour
 - trying to run the module with no `include_metric_filters` and no `exclude_metric_filters` added to the configuration of the module
 - trying to run the module with `include_metric_filters = {}` and/or `exclude_metric_filters = {}` in the configuration of the module
- can you please commit [this change](https://github.com/newrelic/terraform-provider-newrelic/pull/2766/files#r1820915073) I've suggested
- I'd also ask you to squash your commits into one commit which would abide by the conventional commit format, just to make sure the linter doesn't run into issues with the commit

With that, we should be good to proceed with merging the PR :) thanks again for the wonderful contribution!

sjauld commented:
Hi @pranav-new-relic, I have run those tests and all works as expected. I have also added your doc changes (plus aligned the other variables to use the same formatting) and squashed the commits. Thanks!

pranav-new-relic commented:
@sjauld thanks a ton :) we should be good to go. I'll have this PR merged shortly.

pranav-new-relic reviewed (approved):
Approved
